### DISCUSSION TITLE
How to use Heat and HarvestFile to generate 32-bit components

### DISCUSSION THREAD

**Discussion by djohnst10** [wixtoolset/issues#7442] [Questions] [answered]
Is there a way to use Heat and HarvestFile to harvest 32-bit COM DLLs as 32-bit components (`Bitness="always32"`) in a 32-bit directory? I am getting this when I do a build:

```
error WIX0204: ICE80: This 64BitComponent component.dll uses 32BitDirectory AppBinFolder
```

I am aware that I could use `<Transform>` to all the `Bitness="always32"` attribute but I was hoping for a simpler way.

Also, what makes a directory a 32BitDirectory? Is it that it is below the ProgramFilesFolder? The build is being done in Visual Studio 2022 as  x64.

**Comment by BMurri** [comment, marked as answer]
[This page](https://learn.microsoft.com/windows/win32/msi/programfiles64folder) indicates that `ProgramFilesFolder` is ALWAYS a 32-bit directory and `ProgramFiles64Folder` (on the platforms that support it) is ALWAYS a 64-bit directory.

This is the reason that WiX v4 created the `ProgramFiles6432Folder` that will target the bitness of the package itself.
